packages/ember/tests/component_registration_test.js

```diff
@@ -1,200 +1,172 @@
-    appInstance.register('template:components/borf-snorlax', compile('goodfreakingTIMES{{yield}}'));
-  });
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>');
+    this.addTemplate('foo-bar-baz', '{{text}}-{{yield}}');
+    this.add('controller:application', Controller.extend({
+      text: 'outer'
+    }));
 
-  equal(jQuery('#wrapper').text(), 'hello world goodfreakingTIMES-goodfreakingTIMES!!!', 'The component is composed correctly');
-});
+    this.addComponent('my-component', {
+      ComponentClass: Component.extend({
+        text: 'inner',
+        layoutName: 'foo-bar-baz'
+      })
+    });
 
-QUnit.test('Assigning layoutName to a component should setup the template as a layout', function() {
-  expect(1);
+    this.visit('/').then(() => {
+      let text = this.$('#wrapper').text().trim();
+      assert.equal(text, 'inner-outer', 'The component is composed correctly');
+    });
+  }
 
-  setTemplate('application', compile('<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>'));
-  setTemplate('foo-bar-baz', compile('{{text}}-{{yield}}'));
+  ['@test Assigning layoutName and layout to a component should use the `layout` value'](assert) {
+    assert.expect(1);
 
-  boot(() => {
-    appInstance.register('controller:application', Controller.extend({
-      'text': 'outer'
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>');
+    this.addTemplate('foo-bar-baz', 'No way!');
+    this.add('controller:application', Controller.extend({
+      text: 'outer'
     }));
 
-    appInstance.register('component:my-component', Component.extend({
-      text: 'inner',
-      layoutName: 'foo-bar-baz'
-    }));
-  });
+    this.addComponent('my-component', {
+      ComponentClass: Component.extend({
+        text: 'inner',
+        layoutName: 'foo-bar-baz',
+        layout: this.compile('{{text}}-{{yield}}')
+      })
+    });
 
-  equal(jQuery('#wrapper').text(), 'inner-outer', 'The component is composed correctly');
-});
+    this.visit('/').then(() => {
+      let text = this.$('#wrapper').text().trim();
+      assert.equal(text, 'inner-outer', 'The component is composed correctly');
+    });
+  }
 
-QUnit.test('Assigning layoutName and layout to a component should use the `layout` value', function() {
-  expect(1);
+  ['@test Assigning defaultLayout to a component should set it up as a layout if no layout was found [DEPRECATED]'](assert) {
+    assert.expect(2);
 
-  setTemplate('application', compile('<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>'));
-  setTemplate('foo-bar-baz', compile('No way!'));
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>');
 
-  boot(() => {
-    appInstance.register('controller:application', Controller.extend({
-      'text': 'outer'
+    this.add('controller:application', Controller.extend({
```

Comment:
per the original test, this should be registered onto an autobooted application instance

as should `my-component`. Not that you can register on both `application` and on `applicationInstance` and it does have a subtle difference. We should try to recreate the tests accurately here IMO.